scripts/docker_push.sh

```diff
@@ -1,19 +1,21 @@
 #!/bin/bash
 
 # Dockerhub login
-[ ! -z "$DOCKER_PASSWORD" ] || [ ! -z "$DOCKER_USERNAME" ];
-  echo "DOCKER_USERNAME and DOCKER_PASSWORD must be set as environment variables!";
-  exit 1;
-echo "$DOCKER_PASSWORD" | docker login -u "$DOCKER_USERNAME" --password-stdin
-
-# if the build is for a pushed tag, use that tag for the Docker image;
-# otherwise, use "latest"
-if [[ -z "$TRAVIS_TAG" ]]; then
-  image_release_tag="latest";
+if [[ ! -n "$DOCKER_PASSWORD" || ! -n "$DOCKER_USERNAME" ]] >&2; then
+  echo "DOCKER_USERNAME and DOCKER_PASSWORD must be set as environment variables!"
+  exit 1
 else
-  image_release_tag="$TRAVIS_TAG";
-fi
+  docker login -u "$DOCKER_USERNAME" --password-stdin <<< "$DOCKER_PASSWORD" >&2
```

Comment:
If you have an exit status, I think `if {cmd}`  or `if "$?"` should be sufficient.  You don't have to pass only test constructs to conditional constructs.

You can also elide simple pass/fail cases by using `some_cmd || fail_cmd`, or in this case, `docker login -u "$DOCKER_USERNAME" --password-stdin <<< "$DOCKER_PASSWORD" || exit 1`